streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -111,6 +111,12 @@ public StateStoreSupplier build() {
 
                                     @Override
                                     public PersistentKeyValueFactory<K, V> windowed(final long windowSize, final long retentionPeriod, final int numSegments, final boolean retainDuplicates) {
+                                        if (numSegments < 1) {
+                                            throw new IllegalArgumentException("numSegments must be positive");
+                                        }
+                                        if (retentionPeriod < 1) {
```

Comment:
Should this not be an even stronger check: `retentionPeriod < windowSize` ? (some below)